uniref-rest/src/test/java/org/uniprot/api/uniref/queue/UniRefAsyncDownloadIntegrationTest.java

```diff
@@ -0,0 +1,147 @@
+package org.uniprot.api.uniref.queue;
+
+import static org.mockito.Mockito.*;
+import static org.uniprot.api.rest.output.UniProtMediaType.FASTA_MEDIA_TYPE;
+import static org.uniprot.api.uniref.utils.UniRefAsyncDownloadUtils.saveEntriesInSolrAndStore;
+import static org.uniprot.api.uniref.utils.UniRefAsyncDownloadUtils.setUp;
+
+import java.util.List;
+import java.util.stream.Stream;
+
+import org.apache.solr.client.solrj.SolrClient;
+import org.junit.jupiter.api.*;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.params.provider.Arguments;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.boot.test.autoconfigure.web.client.AutoConfigureWebClient;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.boot.test.mock.mockito.SpyBean;
+import org.springframework.context.annotation.PropertySource;
+import org.springframework.http.MediaType;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+import org.springframework.web.client.RestTemplate;
+import org.uniprot.api.rest.controller.AbstractAsyncDownloadIT;
+import org.uniprot.api.rest.download.AsyncDownloadTestConfig;
+import org.uniprot.api.rest.download.configuration.RedisConfiguration;
+import org.uniprot.api.rest.download.queue.AbstractMessageListener;
+import org.uniprot.api.rest.request.DownloadRequest;
+import org.uniprot.api.uniref.UniRefRestApplication;
+import org.uniprot.api.uniref.controller.UniRefDownloadController;
+import org.uniprot.api.uniref.repository.DataStoreTestConfig;
+import org.uniprot.api.uniref.repository.UniRefQueryRepository;
+import org.uniprot.api.uniref.repository.store.UniRefStoreConfig;
+import org.uniprot.api.uniref.request.UniRefDownloadRequest;
+import org.uniprot.core.uniref.UniRefEntryLight;
+import org.uniprot.store.datastore.UniProtStoreClient;
+import org.uniprot.store.search.SolrCollection;
+
+@ActiveProfiles(profiles = {"offline", "asyncDownload", "integration"})
+@EnableConfigurationProperties
+@PropertySource("classpath:application.properties")
+@ContextConfiguration(
+        classes = {
+            DataStoreTestConfig.class,
+            UniRefRestApplication.class,
+            UniRefStoreConfig.class,
+            AsyncDownloadTestConfig.class,
+            RedisConfiguration.class
+        })
+@ExtendWith(SpringExtension.class)
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+@WebMvcTest({UniRefDownloadController.class})
+@AutoConfigureWebClient
+public class UniRefAsyncDownloadIntegrationTest extends AbstractAsyncDownloadIT {
```

Comment:
Optional: Rename class to UniRefAsyncDownloadIT (Just to keep naming..  it helps to search class names)